states/call.go

```diff
@@ -0,0 +1,170 @@
+
+	"github.com/congqixia/birdwatcher/models"
+	"github.com/congqixia/birdwatcher/proto/v2.0/commonpb"
+	"github.com/congqixia/birdwatcher/proto/v2.0/milvuspb"
+	"github.com/congqixia/birdwatcher/proto/v2.0/querypb"
+	"github.com/spf13/cobra"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"google.golang.org/grpc"
+)
+
+func getCallCmd(state State, cli *clientv3.Client, basePath string) *cobra.Command {
+	callCmd := &cobra.Command{
+		Use:   "call",
+		Short: "enter state that could call grpc service of component",
+	}
+
+	callCmd.AddCommand(
+		getCallQueryNodeCmd(state, cli, basePath),
+	)
+
+	return callCmd
+}
+
+func getCallQueryNodeCmd(state State, cli *clientv3.Client, basePath string) *cobra.Command {
+	callQNCmd := &cobra.Command{
+		Use:   "querynode {nodeid}",
+		Short: "components of querynodes",
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) < 1 {
+				cmd.Usage()
+				return
+			}
+			id, err := strconv.ParseInt(args[0], 10, 64)
+			if err != nil {
+				cmd.Usage()
+				return
+			}
+			sessions, err := listSessions(cli, basePath)
+			if err != nil {
+				fmt.Println("failed to list session, err:", err.Error())
+				return
+			}
+
+			for _, session := range sessions {
+				if id == session.ServerID && session.ServerName == "querynode" {
+					//client :=
+					opts := []grpc.DialOption{
+						grpc.WithInsecure(),
+						grpc.WithBlock(),
+						grpc.WithTimeout(2 * time.Second),
+					}
+
+					conn, err := grpc.DialContext(context.Background(), session.Address, opts...)
+					if err != nil {
+						fmt.Printf("failed to connect to proxy(%d) addr: %s, err: %s\n", id, session.Address, err.Error())
+					}
+
+					client := querypb.NewQueryNodeClient(conn)
+					state.SetNext(getQueryNodeState(client, conn, state, session))
+					return
+				}
+			}
+
+			fmt.Printf("querynode id:%d not found\n", id)
+		},
+	}
+
+	return callQNCmd
+}
+
+type queryNodeState struct {
+	cmdState
+	client    querypb.QueryNodeClient
+	conn      *grpc.ClientConn
+	prevState State
+}
+
+func getQueryNodeState(client querypb.QueryNodeClient, conn *grpc.ClientConn, prev State, session *models.Session) State {
+	cmd := &cobra.Command{}
+
+	state := &queryNodeState{
+		cmdState: cmdState{
+			label:   fmt.Sprintf("QueryNode-%d(%s)", session.ServerID, session.Address),
+			rootCmd: cmd,
+		},
+		client: client,
+		conn:   conn,
+	}
+
+	cmd.AddCommand(
+		// GetSegmentInfo collection_id
+		getQNGetSegmentsCmd(client),
+		// GetMetrics
+		getQNGetMetrics(client),
+
+		getBackCmd(state, prev),
+		// exit
+		getExitCmd(state),
+	)
+	return state
+}
+
+func getBackCmd(state, prev State) *cobra.Command {
+	return &cobra.Command{
+		Use: "back",
+		Run: func(cmd *cobra.Command, args []string) {
+			state.SetNext(prev)
+		},
+	}
+}
+
+func getQNGetSegmentsCmd(client querypb.QueryNodeClient) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "GetSegmentInfo collection_id",
+		Short: "call QueryNode.GetSegmentInfo to list segments loaded for collection",
```

Comment:
"call QueryNode.GetSegmentInfo" may confused normal use who haven't touch the milvus code.